header.go

```diff
@@ -152,3 +153,44 @@ func (h Header) String() string {
 		h.Size, h.ProtocolVersion, h.ProfileVersion, h.DataSize, string(h.DataType[:]), h.CRC,
 	)
 }
+
+// MarshalBinary implements the encoding.BinaryMarshaler interface.
+func (h Header) MarshalBinary() ([]byte, error) {
+	buf := &bytes.Buffer{}
+
+	err := binary.Write(buf, binary.LittleEndian, h.Size)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buf, binary.LittleEndian, h.ProtocolVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buf, binary.LittleEndian, h.ProfileVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buf, binary.LittleEndian, h.DataSize)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buf, binary.LittleEndian, h.DataType)
+	if err != nil {
+		return nil, err
+	}
+
+	h.CRC = dyncrc16.Checksum(buf.Bytes())
+
+	if h.Size == 14 {
```

Comment:
Can we use `headerSizeCRC` from `consts.go`?

Reply:
Done